stdlib/public/SDK/Foundation/Foundation.swift

```diff
@@ -127,6 +134,80 @@ extension String : _ObjectiveCBridgeable {
 // Numbers
 //===----------------------------------------------------------------------===//
 
+@_silgen_name("_swift_Foundation_TypePreservingNSNumberWithInt")
+internal func _swift_Foundation_TypePreservingNSNumberWithInt(
+  _ value: Int
+) -> NSNumber
+
+@_silgen_name("_swift_Foundation_TypePreservingNSNumberWithUInt")
+internal func _swift_Foundation_TypePreservingNSNumberWithUInt(
+  _ value: UInt
+) -> NSNumber
+
+@_silgen_name("_swift_Foundation_TypePreservingNSNumberWithFloat")
+internal func _swift_Foundation_TypePreservingNSNumberWithFloat(
+  _ value: Float
+) -> NSNumber
+
+@_silgen_name("_swift_Foundation_TypePreservingNSNumberWithDouble")
+internal func _swift_Foundation_TypePreservingNSNumberWithDouble(
+  _ value: Double
+) -> NSNumber
+
+@_silgen_name("_swift_Foundation_TypePreservingNSNumberWithCGFloat")
+internal func _swift_Foundation_TypePreservingNSNumberWithCGFloat(
+  _ value: CGFloat
+) -> NSNumber
+
+@_silgen_name("_swift_Foundation_TypePreservingNSNumberWithBool")
+internal func _swift_Foundation_TypePreservingNSNumberWithBool(
+  _ value: UInt8
+) -> NSNumber
+
+@_silgen_name("_swift_Foundation_TypePreservingNSNumberGetKind")
+internal func _swift_Foundation_TypePreservingNSNumberGetKind(
+  _ value: NSNumber
+) -> UInt32
+
+internal enum _SwiftTypePreservingNSNumberTag : Int {
```

Comment:
FoundationShims so far has been for Foundation APIs exposed to the core standard library.  I'll add a comment.